SetupDataPkg/SetupDataDxe/SchemaXmlHashDriver/SchemaXmlHash.c

```diff
@@ -0,0 +1,54 @@
+/** @file SchemaXmlHash.c
+  Set config xml hash to BIOS variable
+
+  Copyright (c) Microsoft Corporation.
+
+**/
+#include <Base.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseLib.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+#include <ConfigStdStructDefs.h>
+#include <Generated/ConfigClientGenerated.h>
+#include <Generated/ConfigDataGenerated.h>
+#include <Generated/ConfigProfilesGenerated.h>
+
+/**
+  Module entry point that register to set PlatformConfigData.xml hash variable
+
+  @param FileHandle                     The image handle.
+  @param PeiServices                    The PEI services table.
+
+  @retval Status                        From internal routine or boot object, should not fail
+**/
+EFI_STATUS
+EFIAPI
+SchemaXmlHashDxeEntry (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  EFI_STATUS  Status;
+  EFI_GUID schema_xml_hash_guid = SCHEMA_XML_HASH_GUID;
+
+  DEBUG((DEBUG_ERROR, "In %a\n", __FUNCTION__));
+
+  // Set variable "SCHEMA_XML_HASH" to with data SCHEMA_XML_HASH
+  Status = gRT->SetVariable (
+                  L"SCHEMA_XML_HASH",
+                  &schema_xml_hash_guid,
+                  EFI_VARIABLE_NON_VOLATILE | EFI_VARIABLE_BOOTSERVICE_ACCESS | EFI_VARIABLE_RUNTIME_ACCESS,

```

Comment:
I think this can be a made a volatile variable (i.e. removing `EFI_VARIABLE_NON_VOLATILE` here) so that if the FW is updated, etc., we don't end up with a stale variable in storage.

@apop5 thoughts?

Reply:
Agreed, there is no reason to make this NV.